### PR TITLE
Update for new asset behavior (incl. README updates)

### PR DESCRIPTION
**Pull request by EvanJRichard**
## Summary
recent changes in `go-algorand` did things like: add support for `URL`s and `MetadataHash`es to assets. Also, the `creator` is no longer needed to index into assets. This PR proposes to update the `go` lang SDK to support these changes.
The `README.md` is also updated to discuss Assets.

### Testing
Unit tests updated.

###  See also
https://github.com/algorand/go-algorand-sdk/issues/63
https://github.com/algorand/go-algorand-sdk/issues/64
https://github.com/algorand/go-algorand-sdk/issues/67

### CODE REVIEW

**Comment by EvanJRichard** [comment]
_@rotemh wrote:_
> Does this PR contain the new goldens?

Yes. As part of consolidating all the goldens I'm going to double check/cross-reference/etc. but these should be right. :+1:

File: transaction/transaction.go [hunk 1]
```diff
@@ -326,15 +336,7 @@ func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64
...
-	}
-
-	tx.ConfigAsset = types.AssetID{
-		Creator: creatorAddr,
-		Index:   index,
-	}
+	tx.ConfigAsset = types.AssetIndex(index)
 
 	if newManager != "" {
 		tx.Type = types.AssetConfigTx
```
Comment (justicz) [inline transaction/transaction.go]: I believe this line is not necessary

File: transaction/transaction.go [hunk 2]
```diff
@@ -297,10 +308,9 @@ func MakeAssetCreateTxn(account string, feePerByte, firstRound, lastRound uint64
 // - note is an arbitrary byte array
 // - genesis id corresponds to the id of the network
 // - genesis hash corresponds to the base64-encoded hash of the genesis of the network
-// - creator the address of the asset creator
 // - index is the asset index id
 // - for newManager, newReserve, newFreeze, newClawback see asset.go
-func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64, note []byte, genesisID string, genesisHash string, creator string,
+func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64, note []byte, genesisID, genesisHash string,
```
[the remark this answers is not in the document]
Comment (EvanJRichard) [reply transaction/transaction.go]: Looks like they can't be changed, so I'll leave them out here.

File: transaction/transaction.go (hunk 1 above)
Comment (EvanJRichard) [reply transaction/transaction.go]: It is necessary - without it, compilation fails with `cannot use index (type uint64) as type types.AssetIndex in assignment`

File: client/algod/models/models.go [hunk 3]
```diff
@@ -145,6 +145,19 @@ type AssetParams struct {
...
+	//
+	// required: false
+	URL string `json:"url"`
+
+	// MetadataHash specifies a commitment to some unspecified asset
+	// metadata. The format of this metadata is up to the application.
+	//
+	// required: false
+	// swagger:strfmt byte
+	MetadataHash []byte `json:"metadatahash"`
```
Comment (rotemh) [inline client/algod/models/models.go]: Note that these have changed a bit (nothing too extreme). refer to https://github.com/algorand/go-algorand/blob/master/data/basics/userBalance.go#L206

File: transaction/transaction_test.go [hunk 4]
```diff
@@ -132,8 +132,10 @@ func TestMakeAssetCreateTxn(t *testing.T) {
 	const clawback = addr
 	const unitName = "tst"
 	const assetName = "testcoin"
+	const testURL = "website"
+	const metadataHash = "fACPO4nRgO55j1ndAK3W6Sgc4APkcyFh"
```
Comment (rotemh) [inline transaction/transaction_test.go]: is that a real 32byte hash?
While the code should not reject a shorter than 32 bytes string (If someone decides to make another use of the field, I'm ok with it), it should reject longer ones. 
Also, for the sake of completeness, if this is not a 32 byte string, I'd use a real one.

**Pull request merged**